games/TicTacToe/contract/src/main.sw

```diff
@@ -0,0 +1,98 @@
+contract;
+
+use core::*;
+use std::{address::Address, chain::auth::msg_sender, hash::sha256, storage::{StorageMap, get, store}};
+
+use data_structures::{Game, Players, Winner, core::ops::Eq};
+
+storage {
+    game: Game,
+    player_turn: u64,
+    map: StorageMap<u64,
+    u64>, 
+}
+
+abi TicTacToe {
+    #[storage(write)]fn new_game(player_one: Players, player_two: Players) -> Game;
+    #[storage(read, write)]fn make_move(position: u64);
+    #[storage(read, write)]fn next_player();
+    #[storage(write)]fn end_game(game: Game) -> Winner;
+    #[storage(read)]fn map_is_full() -> bool;
+}
+
+impl TicTacToe for Contract {
+    #[storage(write)]fn new_game(player_one: Players, player_two: Players) -> Game {
+        let mut game = Game {
+            PlayerOne: player_one,
+            PlayerTwo: player_two,
+            winner: Winner::None,
+        };
+        storage.player_turn = 1;
+        let mut counter = 0;
+        while counter < 9 {
+            counter += 1;
+            insert_into_map(counter, 0)
+        }
+        return game;
+    }
+
+    // This function first checks who's turn it is and then inserts to the storage map the player key + the position
+    #[storage(read, write)]fn make_move(position: u64) {
+        if storage.player_turn == 1 {
+            insert_into_map(1, position);
+        }
+        if storage.player_turn == 2 {
+            insert_into_map(2, position);
+        }
```

Comment:
Don't forget to call `next_player` at the end of this function to flip the turn counter!